lib/Sema/CSBindings.cpp

```diff
@@ -1629,11 +1669,34 @@ void BindingSet::dump(llvm::raw_ostream &out, unsigned indent) const {
     attributes.push_back("delayed");
   if (isSubtypeOfExistentialType())
     attributes.push_back("subtype_of_existential");
-  auto literalKind = getLiteralKind();
+  auto literalKind = getLiteralForScore();
   if (literalKind != LiteralBindingKind::None) {
-    auto literalAttrStr = ("[literal: " + getLiteralBindingKind(literalKind)
-                        + "]").str();
-    attributes.push_back(literalAttrStr);
+    std::vector<std::string> kinds;
+    std::string literalAttrStr;
```

Comment:
One last thing (I promise!) - Let's consider whether we really want `literalAttrStr` as storage here... Shouldn't we just send the strings directly into `out` instead after `attributes` are printed? I think this could simplify code in here a bit because you wouldn't need to interleave `kinds` into a string first just to be able to print them as "attributes".